platforms/core-execution/execution/src/test/groovy/org/gradle/internal/execution/history/impl/OriginMetadataSerializerTest.groovy

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.history.impl
+
+import org.gradle.caching.internal.origin.OriginMetadata
+import org.gradle.internal.hash.HashCode
+import org.gradle.internal.serialize.SerializerSpec
+
+import java.time.Duration
+
+class OriginMetadataSerializerTest extends SerializerSpec {
+
+    def serializer = new OriginMetadataSerializer()
+
+    def "can serialize and deserialize origin metadata"() {
+        def originMetadata = new OriginMetadata("test-invocation-id", buildCacheKey, Duration.ofSeconds(123))
+
+        expect:
+        serialize(originMetadata, serializer) == originMetadata
+
+        where:
+        buildCacheKey << [HashCode.fromString("1234567890abcdef"), null]
```

Comment:
💅 We have `TestHashCodes` for this purpose. That makes sure we actually generate the right length of `HashCode`. If you want, you could add a version with a `String` input, but I think using the `long` parameter, or even adding a method that just generates a random one, is better.

Reply:
Using `TestHashCodes` now.